lib/Parse/ParseExpr.cpp

```diff
@@ -2806,14 +2811,22 @@ ExprResult Parser::ParseInteropTypeAnnotation(const Declarator &D, bool IsReturn
 
 ExprResult Parser::ParseBoundsExpressionOrInteropType(const Declarator &D,
                                                       bool IsReturn) {
-  if (StartsBoundsExpression(Tok))
-    return ParseBoundsExpression();
+  ExprResult Result(true);
+  bool isBoundsorItype = false;
 
-  if (StartsInteropTypeAnnotation(Tok))
-    return ParseInteropTypeAnnotation(D, IsReturn);
+  if (StartsBoundsExpression(Tok)) {
+    Result = ParseBoundsExpression();
+    isBoundsorItype = true;
+  }
 
-  Diag(Tok, diag::err_expected_bounds_expr_or_interop_type);
-  return ExprError();
+  if (StartsInteropTypeAnnotation(Tok)) {
```

Comment:
I think you want this code to run only if `StartsBoundsExpression(Tok)` was false.

Otherwise we will parse something like:
bounds(e1, e2) itype(t)

Line 2820 can have the form:
```
} else if (StartsInteropTypeAnnotation(Tok))
```